File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int i = 0;
        int j = 0;
        int maxx = INT_MIN;

        while(j < prices.size()){
            int profit = prices[j] - prices[i];
            maxx = max(maxx, profit);
            if(profit < 0){
                i++;
            }else{
                j++;
            }
        }
        return maxx;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minday = prices[0]
        maxprofit = 0
        for i in range(1, len(prices)):
            profit = prices[i] - minday
            maxprofit = max(maxprofit, profit)
            if prices[i] < minday:
                minday = prices[i]
        
        return maxprofit